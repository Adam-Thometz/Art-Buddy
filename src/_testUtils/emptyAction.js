const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION