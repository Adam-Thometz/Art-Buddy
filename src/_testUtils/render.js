import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
