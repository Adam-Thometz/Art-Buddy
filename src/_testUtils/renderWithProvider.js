import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;