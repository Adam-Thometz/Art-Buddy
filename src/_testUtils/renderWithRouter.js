import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;