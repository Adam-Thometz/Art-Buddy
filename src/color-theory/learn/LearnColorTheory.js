import { useNavigate } from "react-router-dom";

import "./LearnColorTheory.css";

import Icon from "_components/icon/Icon";
import Options from "_components/option/Options";
import WindowNavbar from "_components/window-nav/WindowNavbar";
import { colorTheoryUrls } from "_routes/routeUrls";

import primary from "_media/color-theory/primary.png";
import secondary from "_media/color-theory/secondary.png";
import tertiary from "_media/color-theory/tertiary.png";

const Learn = () => {
  const navigate = useNavigate();

  const goToPrimary = () => navigate(`${colorTheoryUrls.learn}/primary`);
  const goToSecondary = () => navigate(`${colorTheoryUrls.learn}/secondary`);
  const goToTertiary = () => navigate(`${colorTheoryUrls.learn}/tertiary`);

  return (
    <>
      <WindowNavbar page="COLOR THEORY" />
      <main className="LearnColorTheory">
        <span>Learn</span>
        <br />
        <Options width="80%">
          <Icon largeFont icon={primary} text="PRIMARY COLORS" onClick={goToPrimary} />
          <Icon largeFont icon={secondary} text="SECONDARY COLORS" onClick={goToSecondary} />
          <Icon largeFont icon={tertiary} text="TERTIARY COLORS" onClick={goToTertiary} />
        </Options>
      </main>
    </>
  );
};

export default Learn;
