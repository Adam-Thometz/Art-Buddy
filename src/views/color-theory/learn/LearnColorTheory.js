import { useNavigate } from "react-router-dom";

import "./LearnColorTheory.css";

import Icon from "components/icon/Icon";
import Options from "components/option/Options";
import WindowNavbar from "components/window-nav/WindowNavbar";
import { colorTheoryUrls } from "routes/routeUrls";

import { PRIMARY_ICON, SECONDARY_ICON, TERTIARY_ICON } from "assets/color-theory/colorTheory.assets";

const Learn = () => {
  const navigate = useNavigate();

  const goToPrimary = () => navigate(`${colorTheoryUrls.learn}/primary`);
  const goToSecondary = () => navigate(`${colorTheoryUrls.learn}/secondary`);
  const goToTertiary = () => navigate(`${colorTheoryUrls.learn}/tertiary`);

  return (
    <>
      <WindowNavbar page="COLOR THEORY" />
      <main className="LearnColorTheory">
        <span>Learn</span>
        <br />
        <Options width="80%">
          <Icon largeFont icon={PRIMARY_ICON} text="PRIMARY COLORS" onClick={goToPrimary} />
          <Icon largeFont icon={SECONDARY_ICON} text="SECONDARY COLORS" onClick={goToSecondary} />
          <Icon largeFont icon={TERTIARY_ICON} text="TERTIARY COLORS" onClick={goToTertiary} />
        </Options>
      </main>
    </>
  );
};

export default Learn;
