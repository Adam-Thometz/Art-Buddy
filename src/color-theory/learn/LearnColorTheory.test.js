import LearnColorTheory from "./LearnColorTheory";

import { render } from "_testUtils/render";

describe('LearnColorTheory component', () => {
  it('renders without crashing', () => {
    render(<LearnColorTheory />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<LearnColorTheory />);
    expect(asFragment()).toMatchSnapshot();
  });
});