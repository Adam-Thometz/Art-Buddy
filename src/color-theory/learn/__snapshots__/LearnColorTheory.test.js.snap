// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnColorTheory component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="WindowNavbar-games"
  >
    <h1
      class="WindowNavbar-label"
    >
      COLOR THEORY
    </h1>
  </header>
  <main
    class="LearnColorTheory"
  >
    <span>
      Learn
    </span>
    <br />
    <section
      class="Options"
      style="width: 80%;"
    >
      <figure
        class="Icon"
        data-testid=""
      >
        <img
          alt=""
          src="https://art-buddy.s3.amazonaws.com/color-theory/primary.png"
          style="height: 250px; width: 250px; cursor: pointer;"
        />
        <figcaption
          class="Icon-text"
          style="font-size: 2.5rem;"
        >
          PRIMARY COLORS
        </figcaption>
      </figure>
      <figure
        class="Icon"
        data-testid=""
      >
        <img
          alt=""
          src="https://art-buddy.s3.amazonaws.com/color-theory/secondary.png"
          style="height: 250px; width: 250px; cursor: pointer;"
        />
        <figcaption
          class="Icon-text"
          style="font-size: 2.5rem;"
        >
          SECONDARY COLORS
        </figcaption>
      </figure>
      <figure
        class="Icon"
        data-testid=""
      >
        <img
          alt=""
          src="https://art-buddy.s3.amazonaws.com/color-theory/tertiary.png"
          style="height: 250px; width: 250px; cursor: pointer;"
        />
        <figcaption
          class="Icon-text"
          style="font-size: 2.5rem;"
        >
          TERTIARY COLORS
        </figcaption>
      </figure>
    </section>
  </main>
</DocumentFragment>
`;
